src/rider/main/kotlin/me/seclerp/rider/plugins/efcore/features/migrations/remove/RemoveLastMigrationAction.kt

```diff
@@ -17,10 +19,35 @@ class RemoveLastMigrationAction : EfCoreAction() {
         if (dialog.showAndGet()) {
             val migrationsClient = intellijProject.getService<MigrationsClient>()
             val commonOptions = getCommonOptions(dialog)
+            val currentDbContextMigrationsList = dialog.availableMigrationsList
+            val migration = currentDbContextMigrationsList.firstOrNull()
 
             executeCommandUnderProgress(intellijProject, "Removing migration...", "Last migration has been removed") {
-                migrationsClient.removeLast(commonOptions)
+                val res = migrationsClient.removeLast(commonOptions)
+
+                deleteMigrationsFolderIfEmpty(migration)
+
+                res
             }
         }
     }
+
+    private fun deleteMigrationsFolderIfEmpty(migration: MigrationInfo?) {
+        val folder = migration?.migrationFolderAbsPath
+
+        if (migration == null || folder == null) return
+
+        val folderIsEmpty = folderIsEmpty(folder)
+
+        if (folderIsEmpty) {
+            File(folder).delete()
+        }
+    }
+
+    private fun folderIsEmpty(folderPath: String): Boolean {
+        val files = File(folderPath).listFiles()
+        val isEmpty = files?.isEmpty() ?: false
+
+        return isEmpty
+    }
```

Comment:
Create a service responsible for such logic. It's not a responsibility of an action class itself